hydra-main/src/main/java/com/addthis/hydra/job/JobExpand.java

```diff
@@ -134,10 +134,14 @@ static void registerExpander(String macroName, Class<? extends JobConfigExpander
         } catch (InstantiationException | IllegalAccessException e) {
             log.warn("Class '" + clazz + "' registered for '" + macroName + "' cannot be initialized: " + e, e);
         } catch (ClassCastException e) {
-            log.warn("Class '" + clazz + "' registered for '" + macroName + "' is not JobConfigExpander but '" + o.getClass() + "'");
+            log.warn("Class '" + clazz + "' registered for '" + macroName + "' is not JobConfigExpander but '" + getObjectClassIfNotNull(o) + "'");
```

Comment:
Maybe it would be better to split the try block into two. A bit more verbose, but it would be clearer which catch blocks are guarding which line. Something something clarity is great.

It's been an awfully long time, so I suggest closing this PR (with gratitude), and then fixing whatever you want from a cleaner commit.